### PR TITLE
Added docs for cancel_authorization endpoint.

### PR DESCRIPTION
**Pull request by zmaglica**
Fixes https://github.com/Automattic/woocommerce-payments-meta/issues/252

#### Changes proposed in this Pull Request

Added cancel authorization endpoint to the docs and resolved small issue with the status.

#### Testing instructions

* Make sure that you can see cancel authorization API docs in the documentation

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7079` or branch name `add/documentation-for-cancel-authorization-endpoint` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/documentation-for-cancel-authorization-endpoint) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 6f9eb97f16a46757704712592f36c2e4fb26e89f
- Build time: 2023-09-07 14:03:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by RadoslavGeorgiev — APPROVED** [review]
LGTM :shipit:

File: includes/admin/class-wc-rest-payments-orders-controller.php [hunk 1]
```diff
@@ -499,7 +499,7 @@ public function cancel_authorization( WP_REST_Request $request ) {
 
 			$result = $this->gateway->cancel_authorization( $order );
 
-			if ( Intent_Status::SUCCEEDED !== $result['status'] ) {
+			if ( Intent_Status::CANCELED !== $result['status'] ) {
```
Comment (naman03malhotra) [inline includes/admin/class-wc-rest-payments-orders-controller.php]: @zmaglica can you please explain the reasoning behind this change?

**Comment by naman03malhotra** [comment]
I am not sure we should mix the fix and doc in the same PR, we should create a new issue to address the https://github.com/Automattic/woocommerce-payments/pull/7079#discussion_r1318280636 fix you did.

**Comment by zmaglica** [comment]
> I am not sure we should mix the fix and doc in the same PR, we should create a new issue to address the https://github.com/Automattic/woocommerce-payments/pull/7079#discussion_r1318280636 fix you did.

I will address this in another issue.

**Review by naman03malhotra — APPROVED** [review]
LGTM!

**Comment by naman03malhotra** [comment]
@zmaglica will it be deployed with the rest of the API docs, mentioned [here](https://automattic.github.io/woocommerce-payments/#capture-a-terminal-payment)?

**Comment by zmaglica** [comment]
> @zmaglica will it be deployed with the rest of the API docs, mentioned [here](https://automattic.github.io/woocommerce-payments/#capture-a-terminal-payment)?

It should. There is a github actions that scans the docs dir and publishes everything from there.

**Pull request merged**